### PR TITLE
oast: Initial commit

### PR DESCRIPTION
**Pull request by ricekot**
Signed-off-by: ricekot <ricekot@gmail.com>

### CODE REVIEW

File: addOns/oast/oast.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+description = "OAST Support: Exploit Out-Of-Band Vulnerabilities"
+
+zapAddOn {
+    addOnName.set("OAST Support")
+    zapVersion.set("2.10.0")
+
+    manifest {
+        author.set("ZAP Dev Team")
+        url.set("https://www.zaproxy.org/docs/desktop/addons/oast-support/")
```
Comment (kingthorin) [inline addOns/oast/oast.gradle.kts:9]: Is this URL based on what's set here? Or is it based on build file properties?

File: addOns/oast/src/main/javahelp/org/zaproxy/addon/oast/resources/help/contents/oast.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <title>
+        OAST Support
+    </title>
+</head>
+<body>
+<h1>OAST Support</h1>
+
+The OAST Support add-on allows you to detect and exploit out-of-band vulnerabilities in web applications. <br>
+<br>
+OAST stands for Out-of-band Application Security Testing.
```
Comment (kingthorin) [inline addOns/oast/src/main/javahelp/org/zaproxy/addon/oast/resources/help/contents/oast.html:14]: Should we use the full name in the title and H1?

**Review by kingthorin — COMMENTED** [review]
Looks good overall

File: addOns/oast/oast.gradle.kts (hunk 1 above)
Comment (thc202) [reply addOns/oast/oast.gradle.kts:9]: The later part is based on the main help TOC item (`OAST Support` which after normalisation is `oast-support`).

Comment (kingthorin) [reply addOns/oast/oast.gradle.kts:9]: Oh Okay. Thanks

File: addOns/oast/src/main/javahelp/org/zaproxy/addon/oast/resources/help/contents/oast.html (hunk 2 above)
Comment (ricekot) [reply addOns/oast/src/main/javahelp/org/zaproxy/addon/oast/resources/help/contents/oast.html:14]: Works for me. If everyone agrees I can update it.

Comment (thc202) [reply addOns/oast/src/main/javahelp/org/zaproxy/addon/oast/resources/help/contents/oast.html:14]: Sounds good to me.

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/ExtensionOast.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.oast;
+
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+
+public class ExtensionOast extends ExtensionAdaptor {
```
Comment (thc202) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/ExtensionOast.java:24]: Should override `canUnload`.

Comment (thc202) [reply addOns/oast/src/main/java/org/zaproxy/addon/oast/ExtensionOast.java:24]: btw, it could add the API as well.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**